tests/conftest.py

```diff
@@ -23,6 +23,15 @@
     pass  # so the environment without spark doesn't break
 
 
+@pytest.fixture
+def tmp_dir(tmp_path_factory):
```

Comment:
a note of naming, the [pytest temporary dir](https://docs.pytest.org/en/latest/tmpdir.html#the-tmpdir-fixture) is called `tmpdir` which is almost the same as ours. Maybe we should rename it to avoid users mistake pytest fixture with our fixture. Maybe `temporary_dir`?

Reply:
good suggestion. maybe even more specific, something like `fn_scope_tmpdir`?

Reply:
or does it sound like some kind of function because of some name convention?

Reply:
i'm missing where tmp_path_factory is coming from?

Reply:
@gramhagen it is pytest's fixture: https://docs.pytest.org/en/latest/tmpdir.html

Reply:
name changed to just `tmp`, named after linux tmp dir. I picked that for simplicity, but if you think it is not very descriptive, feel free to let me know or give any suggestions. `temporary_dir` looks (visually) unbalanced to me...haha

Reply:
one question, do we foresee any case where we want to have a directory per (pytest) module or even per (pytest) session?

Reply:
We can make fixtures for them when we need, like `sess_tmp` and `module_tmp`.